services/worker/tests/job_runners/split/test_duckdb_index.py

```diff
@@ -0,0 +1,234 @@
+                    "triggered_by": ["config-parquet", "config-split-names-from-streaming"],
+                },
+            }
+        )
+        return SplitDuckDbIndexJobRunner(
+            job_info={
+                "type": SplitDuckDbIndexJobRunner.get_job_type(),
+                "params": {
+                    "dataset": dataset,
+                    "revision": "revision",
+                    "config": config,
+                    "split": split,
+                },
+                "job_id": "job_id",
+                "priority": Priority.NORMAL,
+            },
+            app_config=app_config,
+            processing_step=processing_graph.get_processing_step(processing_step_name),
+            duckdb_index_cache_directory=duckdb_index_cache_directory,
+        )
+
+    return _get_job_runner
+
+
+@pytest.fixture
+def get_parquet_job_runner(
+    libraries_resource: LibrariesResource,
+    cache_mongo_resource: CacheMongoResource,
+    queue_mongo_resource: QueueMongoResource,
+) -> GetParquetJobRunner:
+    def _get_job_runner(
+        dataset: str,
+        config: str,
+        app_config: AppConfig,
+    ) -> ConfigParquetAndInfoJobRunner:
+        processing_step_name = ConfigParquetAndInfoJobRunner.get_job_type()
+        processing_graph = ProcessingGraph(
+            {
+                "dataset-level": {"input_type": "dataset"},
+                processing_step_name: {
+                    "input_type": "config",
+                    "job_runner_version": ConfigParquetAndInfoJobRunner.get_job_runner_version(),
+                    "triggered_by": "dataset-level",
+                },
+            }
+        )
+        return ConfigParquetAndInfoJobRunner(
+            job_info={
+                "type": ConfigParquetAndInfoJobRunner.get_job_type(),
+                "params": {
+                    "dataset": dataset,
+                    "revision": "revision",
+                    "config": config,
+                    "split": None,
+                },
+                "job_id": "job_id",
+                "priority": Priority.NORMAL,
+            },
+            app_config=app_config,
+            processing_step=processing_graph.get_processing_step(processing_step_name),
+            hf_datasets_cache=libraries_resource.hf_datasets_cache,
+        )
+
+    return _get_job_runner
+
+
+@pytest.mark.parametrize(
+    "hub_dataset_name,max_parquet_size_bytes,expected_error_code",
+    [
+        ("duckdb_index", None, None),
+        ("duckdb_index", 1_000, "SplitWithTooBigParquetError"),  # parquet size is 2812
+        ("public", None, "NoIndexableColumnsError"),  # dataset does not have string columns to index
+    ],
+)
+def test_compute(
+    get_parquet_job_runner: GetParquetJobRunner,
+    get_job_runner: GetJobRunner,
+    app_config: AppConfig,
+    hub_responses_public: HubDatasetTest,
+    hub_responses_duckdb_index: HubDatasetTest,
```

Comment:
maybe delete the remote index file, in case another test uses the same repository (fixture scope is session) ?

Reply:
> in case another test uses the same repository

Is it possible?